File: benchmark/go-app/tracing.go
package main

import (
	"context"
	"os"

	"go.opentelemetry.io/otel"
	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
	"go.opentelemetry.io/otel/sdk/resource"
	sdktrace "go.opentelemetry.io/otel/sdk/trace"
	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
	"go.opentelemetry.io/otel/trace"
)

const (
	defaultOtlpEndpoint = "localhost:4317"
)

var (
	tracer trace.Tracer
)

func setupTraceProvider(ctx context.Context) (*sdktrace.TracerProvider, error) {
	otlpEndpoint := os.Getenv("OTLP_ENDPOINT")
	if otlpEndpoint == "" {
		otlpEndpoint = defaultOtlpEndpoint
	}

	insecureOpt := otlptracegrpc.WithInsecure()
	endpointOpt := otlptracegrpc.WithEndpoint(otlpEndpoint)

	exporter, err := otlptracegrpc.New(ctx, insecureOpt, endpointOpt)
	if err != nil {
		return nil, err
	}

	r := resource.NewWithAttributes(
		semconv.SchemaURL,
		semconv.ServiceNameKey.String("go-app"),
	)

	tp := sdktrace.NewTracerProvider(sdktrace.WithBatcher(exporter), sdktrace.WithResource(r))
	otel.SetTracerProvider(tp)

	tracer = tp.Tracer("internal")

	return tp, nil
}

File: benchmark/python-app/tracing.py
import os

from opentelemetry import trace
from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
from opentelemetry.sdk.resources import SERVICE_NAME, Resource
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import BatchSpanProcessor

DEFAULT_OTLP_ENDPOINT = "localhost:4317"
OTLP_ENDPOINT = os.getenv("OTLP_ENDPOINT", DEFAULT_OTLP_ENDPOINT)

# Start configuring OpenTelemetry.
resource = Resource(attributes={SERVICE_NAME: "python-app"})
provider = TracerProvider(resource=resource)

# OpenTelemetry Protocol Exporter (OTLP) Exporter
processor = BatchSpanProcessor(
    OTLPSpanExporter(
        # Change default HTTPS -> HTTP.
        insecure=True,
        # Update default OTLP reciver endpoint.
        endpoint=OTLP_ENDPOINT,
    )
)
provider.add_span_processor(processor)

# Sets the global default tracer provider.
trace.set_tracer_provider(provider)

# Creates a tracer from the global tracer provider.
tracer = trace.get_tracer("internal")
